Helper/AbstractHelper.php

```diff
@@ -46,24 +47,34 @@ public function __construct(
 
     public function getSettingsVal($val, $storeId = null)
     {
-        $scopeType = ScopeInterface::SCOPE_STORES;
-        $scopeCode = null;
-
         if ($storeId) {
-            $scopeCode = $this->storeManager->getStore($storeId)->getCode();
-        } else {
-            $storeCode = $this->_request->getParam('store');
-            $websiteCode = $this->_request->getParam('website');
-            if ($storeCode) {
-                $scopeType = ScopeInterface::SCOPE_STORE;
-                $scopeCode = $storeCode;
-            } elseif ($websiteCode) {
-                $scopeType = ScopeInterface::SCOPE_WEBSITE;
-                $scopeCode = $websiteCode;
-            } else {
-                $scopeCode = $this->storeManager->getStore()->getCode();
-            }
+            return $this->scopeConfig->getValue($val, ScopeInterface::SCOPE_STORE, $this->storeManager->getStore($storeId)->getCode());
+        }
+        
+        if ($storeCode = $this->_request->getParam('store')) {
+            return $this->scopeConfig->getValue($val, ScopeInterface::SCOPE_STORE, $storeCode);
         }
-        return $this->scopeConfig->getValue($val, $scopeType, $scopeCode);
+
+        if ($orderId = $this->_request->getParam('order_id')) {
+            $storeCode = $this->getStoreCodeFromOrderId($orderId);
+            return $this->scopeConfig->getValue($val, ScopeInterface::SCOPE_STORE, $storeCode);
+        }
+
+        if ($websiteCode = $this->_request->getParam('website')) {
+            return $this->scopeConfig->getValue($val, ScopeInterface::SCOPE_WEBSITE, $websiteCode);
+        }
+
+        return $this->scopeConfig->getValue($val, ScopeInterface::SCOPE_STORE, $this->storeManager->getStore()->getCode());
+    }
+
+    private function getStoreCodeFromOrderId($orderId) {
+        /** @var CollectionFactoryInterface $orderFactory */
+        $orderFactory = $this->objectManager->create('\Magento\Sales\Model\ResourceModel\Order\CollectionFactoryInterface');
```

Comment:
Order repository injected as a dependency on the constructor should be used.

Reply:
Are you saying 1) use order repo instead of collection, and 2) use DI?

Reply:
Yes, both 1) and 2), for instance:
```php
public function __construct(
    OrderRepositoryInterface $orderRepository
) {
    $this->orderRepository = $orderRepository;
}

// ...

$order = $this->orderRepository->get($orderId);
```